src/Web/Slack/Experimental/Blocks/Types.hs

```diff
@@ -361,10 +376,34 @@ instance Show SlackAccessory where
 
 -- | Small helper function for constructing a section with a button accessory out of a button and text components
 sectionWithButtonAccessory :: SlackAction -> SlackText -> SlackBlock
-sectionWithButtonAccessory btn txt = SlackBlockSection txt (Just $ SlackButtonAccessory btn)
+sectionWithButtonAccessory btn txt =
+  SlackBlockSection $
+    (slackSectionWithText txt)
+      { slackSectionAccessory = Just $ SlackButtonAccessory btn
+      }
+
+-- | <https://api.slack.com/reference/block-kit/blocks#section>
+data SlackSection = SlackSection
+  { slackSectionText :: Maybe SlackText
+  -- ^ May be absent if 'slackSectionFields' is present.
+  , slackSectionBlockId :: Maybe SlackBlockId
+  , slackSectionFields :: Maybe [SlackText]
+  -- ^ Required if 'slackSectionText' is not provided.
+  , slackSectionAccessory :: Maybe SlackAccessory
+  }
```

Comment:
`slackSectionText` and `slackSectionFields` should really be `SlackTextObject`, but using that type would break the library's API even further. 

The consequence of using `SlackText` is that the library only supports `mrkdwn` text. This is an incomplete implementation of Slack's API, which accepts both `plain_text` and `mrkdwn` for the section `text` and `fields` attributes.